drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/create_referral_job.rb

```diff
@@ -165,35 +163,54 @@ def update_client_addresses(client)
           ),
           AddressID: Hmis::Hud::Base.generate_uuid,
           **common_client_attrs(client),
-        }
+        )
+      end
+
+      # See if any values are equal to existing values. If so, update the existing value.
+      old_addresses = client.addresses
+      new_addreseses.each do |addy|
+        addy.id = old_addresses.find { |old| old.equal_for_merge?(addy) }&.id
       end
-      Hmis::Hud::CustomClientAddress.import!(client_address_attrs)
+
+      Hmis::Hud::CustomClientAddress.import!(
+        new_addreseses,
+        on_duplicate_key_update: { conflict_target: [:id], columns: [:DateUpdated] },
```

Comment:
I found the documentation on this kind of confusing, but it seems to be working. The `columns: [:DateUpdated]` part may not be needed.

Reply:
I believe that means it will only change the DateUpdated column.

Reply:
> One potential problem is that it looks like we are not removing stale records. For example, if new addresses is empty, the current addresses will not be removed. Is that expected?

Yes. The fact that we're receiving new addresses from LINK should not wipe away any addresses that users may have input manually on the user profile. Users can see that date updated, so they can tell which addresses are newest.

The goal here is to add the new addresses to the existing addresses, but without duplicates being created.

Reply:
Alright, this is updated to be much simpler: if it's a duplicate, update the old record. else, import it.